apple/RNGestureHandler.mm

```diff
@@ -495,6 +506,37 @@ - (BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer
   return NO;
 }
 
+- (BOOL)areRecognizersCompatible:(UIGestureRecognizer *)gestureRecognizer
+                 otherRecognizer:(UIGestureRecognizer *)otherGestureRecognizer
+{
+  // Check if otherGestureRecognizer is instance of UIScrollViewPanGestureRecognizer (is UIPanGestureRecognizer and has
+  // scrollView property)
+  if ([otherGestureRecognizer isKindOfClass:[UIPanGestureRecognizer class]] &&
+      [otherGestureRecognizer respondsToSelector:@selector(scrollView)] &&
+      [gestureRecognizer isKindOfClass:[RNDummyGestureRecognizer class]]) {
+    UIView *gestureRecognizerView = gestureRecognizer.view;
+    UIView *otherGestureRecognizerView = otherGestureRecognizer.view;
+
+#ifdef RCT_NEW_ARCH_ENABLED
+    if ([gestureRecognizerView isKindOfClass:[RCTScrollViewComponentView class]]) {
+      UIScrollView *scrollView = ((RCTScrollViewComponentView *)gestureRecognizerView).scrollView;
+      if (scrollView == otherGestureRecognizerView) {
+        return YES;
+      }
+    }
+#else
+    if ([gestureRecognizerView isKindOfClass:[RCTScrollView class]]) {
+      UIScrollView *scrollView = [gestureRecognizerView.subviews objectAtIndex:0];
+      if (scrollView == otherGestureRecognizerView) {
+        return YES;
+      }
+    }
+#endif
```

Comment:
Assuming copying the state from the scroll's recognizer works, we would need that logic repeated the third time. Can we move this to a helper function that would do these checks and return `UIScrollView` if it can be extracted (`nil` otherwise)?